src/util/configurations.rs

```diff
@@ -0,0 +1,83 @@
+//! This is the config struct used to pass in configurations into the channel. This is kept by the channel manager
+//! channel_limits is kept in an arc so that each channel can keep a reference to the parent one in channel manager
+//! channel options is cloned into channel because those are all specified unique channel settings. It uses channel managers one as a base
+use std::sync::Arc;
+
+/// This is the main user configuration
+/// This struct should contain all user customizable options as this is passed to the channel to be accessed
+#[derive(Clone, Debug)]
+pub struct UserConfigurations{
+	/// optional user specified channel limits
+	/// These are only used on startup of channels, and are referenced to a single instance
+    pub channel_limits : ChannelLimits,
+	/// Channel options can change afterwords and are unique to each channel
+	pub channel_options : ChannelOptions,
```

Comment:
ChannelOptions and ChannelLimits are not very descriptive names. Maybe something describing what they're used for - one during initial handshake and one that can (hopefully eventually) change at runtime.